examples/defaulter-gen/generators/defaulter.go

```diff
@@ -520,15 +520,16 @@ func populateDefaultValue(node *callNode, t *types.Type, tags string, commentLin
 	var defaultString string
 	if len(defaultMap) == 1 {
 		defaultString = defaultMap[0]
+	} else if len(defaultMap) > 1 {
+		klog.Fatalf("Found more than one default tag for %v", t.Kind)
 	}
 
-	t, depth := resolveTypeAndDepth(t)
-	if depth > 0 && defaultString == "" {
+	baseT, depth := resolveTypeAndDepth(t)
+	if len(depth) > 0 && defaultString == "" {
 		defaultString = getNestedDefault(t)
 	}
-	if len(defaultMap) > 1 {
-		klog.Fatalf("Found more than one default tag for %v", t.Kind)
-	} else if len(defaultMap) == 0 {
+
+	if len(defaultString) == 0 {
```

Comment:
Changed `len(defaultMap)` -> len(defaultString)` since that seemed to be the original intention to allow nested default to work. Nested default is only used if default map is empty, so the original condition here would block their use.